openpype/modules/deadline/plugins/publish/submit_nuke_deadline.py

```diff
@@ -470,6 +466,21 @@ def expected_files(
         dirname = os.path.dirname(path)
         file = os.path.basename(path)
 
+        # since some files might be already tagged as publish_on_farm
+        # we need to avoid adding them to expected files since those would be
+        # duplicated into metadata.json file
+        representations = instance.data.get("representations", [])
+        # check if file is not in representations with publish_on_farm tag
+        for repre in representations:
+            # Skip if 'publish_on_farm' not available
+            if "publish_on_farm" not in repre.get("tags", []):
+                continue
+            # is file in representations files?
+            if file in repre.get("files", []):
+                self.log.debug(
+                    "Skipping expected file: {}".format(path))
+                return
```

Comment:
`if file in files` will be very slow I assume for representations with a lot of frames, no?

Also, are we sure that `file` is a path like we'd expect. Just below this we're doing `if "#" in file` so file apparently can have `#` as a character that I assume might not be like that in the expected file list, so the comparison is invalid?

I feel like this is _actually_ hinting to another problem that should be solved - the issue sounds like that we're **also** registering `files` in the case where it shouldn't be?

Reply:
Yes, it was bit unclear. I had improved comments and regarding the file check. This will always be single file check so speed is not an issue.